flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/nodes/physical/stream/StreamPhysicalGlobalGroupAggregate.scala

```diff
@@ -123,6 +130,7 @@ class StreamPhysicalGlobalGroupAggregate(
       generateUpdateBefore,
       needRetraction,
       indexOfCountStar.map(Integer.valueOf).orNull,
+      StateTtlHint.getStateTtlFromHintOnSingleRel(hints),
```

Comment:
This is commented on `explainTerms`.
Before `STATE_TTL` hint is introduced, the rest query hints(mainly join hints for batch and lookup join hints for stream) are converted and use `itemIf`
So, I rethink the explanation format. How about
```scala
// for agg
      .itemIf(
        "stateTtl",
        StateTtlHint.getStateTtlFromHintOnSingleRel(hints),
        hints.stream().anyMatch(hint => StateTtlHint.isStateTtlHint(hint.hintName)))

// which yields
// stateTtl=[1d]

// for regular join
val ttlHint = StateTtlHint.getStateTtlFromHintOnBiRel(getHints)
      .itemIf("leftStateTtl", ttlHint.get(0), ttlHint.containsKey(0))
      .itemIf("rightStateTtl", ttlHint.get(1), ttlHint.containsKey(1))
// which yields
// leftStateTtl=[1d], rightStateTtl=[3d]
```

Reply:
I think the printing of state hint should be placed in RelTreeWriterImpl and continue to be managed uniformly by withQueryHint.